indico/modules/events/persons/controllers.py

```diff
@@ -152,21 +148,32 @@ def _checkParams(self, params):
     def _process(self):
         person_ids = request.form.getlist('person_id')
         recipients = {p.email for p in self._find_event_persons(person_ids) if p.email}
-        form = EmailEventPersonsForm(person_id=person_ids, recipients=', '.join(recipients))
+        if 'no_account' in request.args:
+            tpl = get_template_module('events/registration/emails/pending_submitter.html', event=self.event_new)
+            default_body = tpl.get_html_body()
+            form = EmailEventPersonsForm(person_id=person_ids, recipients=', '.join(recipients), body=default_body)
+        else:
+            form = EmailEventPersonsForm(person_id=person_ids, recipients=', '.join(recipients))
         if form.validate_on_submit():
-            for recipient in recipients:
-                email = make_email(to_list=recipient, from_address=form.from_address.data,
-                                   subject=form.subject.data, body=form.body.data, html=True)
-                send_email(email, self.event_new, 'Event Persons')
+            self._send_emails(form, self._find_event_persons(person_ids))
             num = len(recipients)
             flash(ngettext('Your email has been sent.', '{} emails have been sent.', num).format(num))
             return jsonify_data()
         return jsonify_form(form, submit=_('Send'))
 
+    def _send_emails(self, form, recipients):
+        for recipient in recipients:
+            email_body = replace_placeholders('roles-email', form.body.data, person=recipient,
+                                              event=self.event_new)
+            email_subject = replace_placeholders('roles-email', form.subject.data, person=recipient,
+                                                 event=self.event_new)
+            email = make_email(to_list=recipient.email, from_address=form.from_address.data,
+                               body=email_body, subject=email_subject, html=True)
+            send_email(email, self.event_new, 'Event Persons')
+
     def _find_event_persons(self, person_ids):
         return (self.event_new.persons
-                .filter(EventPerson.id.in_(person_ids), EventPerson.email != '')
-                .all())
+                .filter(EventPerson.id.in_(person_ids), EventPerson.email != '').all())
```

Comment:
Please keep this in separate lines, it's more readable to have no more than one chain element in each line when wrapping queries over multiple lines.